File: approx_pi.c
double approx_pi(int intervals)
{
     double pi = 0.0;
     int i;
     for (i = 0; i < intervals; i++) {
         pi += (4 - ((i % 2) * 8)) / (float)(2 * i + 1);
     }
     return pi;
}

File: approx_pi.f90
subroutine approx_pi(intervals, pi)
  integer, intent(in) :: intervals
  double precision, intent(out) :: pi
  integer i
  pi = 0
  do i = 0, intervals - 1
     pi = pi + (4 - (mod(i,2) * 8)) / dble(2 * i + 1)
  enddo
end subroutine approx_pi

